test/e2e/set/scyllacluster/scyllacluster_tls.go

```diff
@@ -0,0 +1,380 @@
+			func() {
+				framework.By("Checkpointing secret %q", item.secretName)
+
+				initialSecret, err := f.KubeClient().CoreV1().Secrets(f.Namespace()).Get(ctx, item.secretName, metav1.GetOptions{})
+				o.Expect(err).NotTo(o.HaveOccurred())
+				o.Expect(initialSecret.Data).To(o.HaveKey("tls.crt"))
+				o.Expect(initialSecret.Data).To(o.HaveKey("tls.key"))
+				o.Expect(initialSecret.Data["tls.crt"]).NotTo(o.BeEmpty())
+				o.Expect(initialSecret.Data["tls.key"]).NotTo(o.BeEmpty())
+				o.Expect(initialSecret.Annotations).To(o.HaveKeyWithValue("certificates.internal.scylla-operator.scylladb.com/refresh-reason", "needs new cert"))
+
+				var configMap *corev1.ConfigMap
+				var initialBundleCert *x509.Certificate
+				if len(item.cmName) != 0 {
+					framework.By("Checkpointing configmap %q", item.secretName)
+					configMap, err = f.KubeClient().CoreV1().ConfigMaps(f.Namespace()).Get(ctx, item.cmName, metav1.GetOptions{})
+					o.Expect(err).NotTo(o.HaveOccurred())
+
+					bundleCerts, err := kubecrypto.GetCABundleFromConfigMap(configMap)
+					o.Expect(err).NotTo(o.HaveOccurred())
+					o.Expect(bundleCerts).To(o.HaveLen(1))
+					initialBundleCert = bundleCerts[0]
+				}
+
+				framework.By("Replacing secret %q cert to be past its latest refresh time (90%%)", item.secretName)
+
+				initialCert, err := kubecrypto.GetCertFromSecret(initialSecret.DeepCopy())
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				cert, key, err := kubecrypto.GetCertKeyFromSecret(initialSecret.DeepCopy())
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				issuerSecret, err := f.KubeClient().CoreV1().Secrets(f.Namespace()).Get(ctx, item.issuerName, metav1.GetOptions{})
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				issuerCert, issuerKey, err := kubecrypto.GetCertKeyFromSecret(issuerSecret)
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				now := time.Now()
+				// We can pick any validity long enough for the test.
+				validity := 10 * time.Hour
+				cert.NotAfter = now.Add(validity / 10)
+				cert.NotBefore = now.Add(-validity / 10 * 9)
+				o.Expect(cert.NotAfter.Sub(cert.NotBefore)).To(o.Equal(validity))
+
+				// For self-signed certs we can reuse the key.
+				cert, err = crypto.SignCertificate(cert, key.Public().(*rsa.PublicKey), issuerCert, issuerKey)
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				o.Expect(cert.AuthorityKeyId).To(o.Equal(initialCert.AuthorityKeyId))
+
+				pemBytes, err := crypto.EncodeCertificates(cert)
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				secret, err := f.KubeClient().CoreV1().Secrets(f.Namespace()).Patch(
+					ctx,
+					initialSecret.Name,
+					types.JSONPatchType,
+					[]byte(fmt.Sprintf(`[{"op": "replace", "path": "/data/tls.crt", "value": %q}]`, base64.StdEncoding.EncodeToString(pemBytes))),
+					metav1.PatchOptions{},
+				)
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				framework.By("Waiting for Secret %q to be updated", secret.Name)
+				waitCtxL1, waitCtxL1Cancel := context.WithTimeout(ctx, 30*time.Second)
```

Comment:
L1? it's a typo or it stands for something?

Reply:
local